### PR TITLE
reward manifest change

### PR DESCRIPTION
**Pull request by andreea-popescu-reef**

### CODE REVIEW

File: validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py [hunk 1]
```diff
@@ -471,7 +522,9 @@ async def _execute_synthetic_job(miner_client, job: SyntheticJob):
             )
 
             # if job passed, save synthetic job score
-            job.score = score
+            job.score = await apply_manifest_incentive(
```
Comment (mzukowski-reef) [inline validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py:525]: I am afraid that doing it here will have impact on batch performance - every ORM call is run in single thread - please check impact on preprod

File: compute_horde/compute_horde/mv_protocol/miner_requests.py [hunk 2]
```diff
@@ -28,6 +28,10 @@ class ExecutorClassManifest(pydantic.BaseModel):
 class ExecutorManifest(pydantic.BaseModel):
     executor_classes: list[ExecutorClassManifest]
 
+    @property
+    def count(self) -> int:
```
Comment (mzukowski-reef) [inline compute_horde/compute_horde/mv_protocol/miner_requests.py:32]: total_count?

File: validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py [hunk 3]
```diff
@@ -471,7 +522,9 @@ async def _execute_synthetic_job(miner_client, job: SyntheticJob):
...
+                miner_client.miner_hotkey, job.batch_id, score
```
Comment (mzukowski-reef) [inline validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py:526]: also different score is sent in receipt

File: validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py [hunk 4]
```diff
@@ -324,7 +334,48 @@ async def execute_miner_synthetic_jobs(batch_id, miner_id, miner_hotkey, axon_in
...
 
 
-async def _execute_synthetic_job(miner_client, job: SyntheticJob):
+async def apply_manifest_incentive(miner_hotkey: str, batch_id: int, score: float) -> float:
+    miner = await Miner.objects.aget(hotkey=miner_hotkey)
+
+    # get last 3 batches and manifests
+    batches = [
+        batch
+        async for batch in SyntheticJobBatch.objects.filter(id__lte=batch_id).order_by("-id")[:3]
```
[the remark this answers is not in the document]
Comment (andreea-popescu-reef) [reply validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py:343]: started_at is not indexed and batch_id is already in increasing order so it should be the same

File: validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py (hunk 1 above)
Comment (andreea-popescu-reef) [reply validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py:526]: ran the debug_run_synthetic_jobs in preprod and seem fine - though there's not much monitoring to check

File: validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py [hunk 5]
```diff
@@ -325,6 +347,55 @@ async def execute_miner_synthetic_jobs(batch_id, miner_id, miner_hotkey, axon_in
...
+            logger.debug(
+                f"Applied manifest incentive for {miner_hotkey} - miner has increased number of executors significantly"
+            )
+        elif len(manifests) < 3:
+            logger.debug(
+                f"Applied manifest incentive for {miner_hotkey} - validator has missed one of the previous 2 synthetic jobs windows"
+            )
+        else:
+            # do not apply manifest incentive - return original score
+            return score
```
Comment (emnoor-reef) [inline validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py:391]: I think this check is missing: "if “this” validator missed last or second last window"

we are getting last 3, which can be more than 2 windows ago if this validator somehow didn't run a batch on a previous window.

**Review by emnoor-reef — APPROVED** [review]

File: validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py (hunk 5 above)
Comment (emnoor-reef) [reply validator/app/src/compute_horde_validator/validator/synthetic_jobs/utils.py:391]: But consider this a bug, that can be solved in the next release.

**Pull request merged**